### PR TITLE
Compile fixes for ITKv5 with legacy disabled

### PR DESCRIPTION
**Pull request by dzenanz**

### CODE REVIEW

File: examples/computeKrcahBoneEnhancement.cxx [hunk 1]
```diff
@@ -64,7 +64,7 @@ int main(int argc, char * argv[])
   double thisSigma;
   itk::Array< double > sigmaArray;
   sigmaArray.SetSize(numberOfSigma);
-  for (unsigned int i = 0; i < numberOfSigma; ++i) {
```
Comment (jhlegarreta) [inline examples/computeKrcahBoneEnhancement.cxx]: May be `numberOfSigma` should be `unsigned`?

**Review by jhlegarreta — COMMENTED** [review]
An in-line comment.

The rest of the changes seem OK to me.

**Comment by jhlegarreta** [comment]
Sorry, although this has been merged, wondering why [this line](https://github.com/InsightSoftwareConsortium/ITKBoneEnhancement/blob/00f62b474a74079d04397bad881b080743f41b9d/examples/computeKrcahBoneEnhancement.cxx#L67) does not throw an `unsigned long` vs. `unsigned int` comparison warning now.

**Comment by dzenanz** [comment]
Because I changed the declaration of `numberOfSigma` from `int` to `ulong`:
https://github.com/InsightSoftwareConsortium/ITKBoneEnhancement/blob/00f62b474a74079d04397bad881b080743f41b9d/examples/computeKrcahBoneEnhancement.cxx#L63
in response to your earlier comment.

**Comment by jhlegarreta** [comment]
Yes, I realized @dzenanz , but by the same token you first proposed to change the `for` loop counter type, I'd expect now an `unsigned long` vs. `unsigned int` warning.

**Comment by dzenanz** [comment]
When two unsigned numbers are compared, the smaller one is extended to the size of the larger one (I think), so no problems and no warning.

**Comment by jhlegarreta** [comment]
OK. Thanks @dzenanz .

**Pull request merged**